### PR TITLE
Add `Record` JSON Storage Format to DG

### PR DESCRIPTION
**Pull request by longnguyentan**
Add `Record` JSON Storage Format to DG
### Key Changes:
- Added a new section in the Developer Guide detailing the JSON representation of the `Record` class.
- Provided an illustrative example with a breakdown of each JSON field for clarity.
- Highlighted default values associated with certain fields when no data is available.

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -22,12 +22,7 @@
 import seedu.address.model.ReadOnlyUserPrefs;
 import seedu.address.model.UserPrefs;
 import seedu.address.model.util.SampleDataUtil;
-import seedu.address.storage.AddressBookStorage;
-import seedu.address.storage.JsonAddressBookStorage;
-import seedu.address.storage.JsonUserPrefsStorage;
-import seedu.address.storage.Storage;
-import seedu.address.storage.StorageManager;
-import seedu.address.storage.UserPrefsStorage;
+import seedu.address.storage.*;
```
Comment (thienmy0) [inline src/main/java/seedu/address/MainApp.java:25]: Perhaps * import statements should be avoided?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -641,3 +641,35 @@ testers are expected to do more *exploratory* testing.
...
+        "address": "311, Clementi Ave 2, #02-25",
+        "tags": ["friend", "owesMoney"]
+    },
+    "initialObservations": "Patient complains of a persistent cough for 2 weeks.",
+    "diagnosis": "Common cold, aggravated due to not taking enough rest.",
+    "treatmentPlan": "Rest, Hydration, and prescribed cough syrup."
+}
+```
+
+#### Explanation:
```
Comment (thienmy0) [inline docs/DeveloperGuide.md:668]: I like how you added the explanation about how you integrated Record into Storage!

File: src/main/resources/view/HelpWindow.css [hunk 3]
```diff
@@ -3,17 +3,48 @@
 }
 
 #copyButton {
-    -fx-background-color: dimgray;
+    -fx-background-color: #3C4EEF;
```
Comment (thienmy0) [inline src/main/resources/view/HelpWindow.css:6]: I like how you adapted our UI to look more like the design Jingya made:)

**Review by thienmy0 — APPROVED** [review]
LGTM! 👍  Mainly changes on DG, UI, integrating Record into Storage & some checkstyle fixing. This PR seems to include the changes made from your other 3 PRs too so might want to check on that.

File: src/main/resources/view/HelpWindow.css (hunk 3 above)
Comment (wujy28) [reply src/main/resources/view/HelpWindow.css:6]: Yeah, thanks for doing this!

File: src/main/java/seedu/address/model/patient/UniqueRecordList.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import javafx.collections.ObservableList;
+
+/**
+ * A list of records that enforces uniqueness between its elements and does not allow nulls.
+ * A record is considered unique by comparing using {@code Record#equals(Record)}.
+ * As such, adding and updating of records uses Record#equals(Record) for equality
+ * to ensure that the record being added or updated is
+ * unique in terms of identity in the UniqueRecordList.
+ */
+public class UniqueRecordList implements Iterable<Record> {
```
Comment (wujy28) [inline src/main/java/seedu/address/model/patient/UniqueRecordList.java:19]: Neat way to store all the records in the system!

**Review by wujy28 — APPROVED** [review]
Nice and clean implementation! LGTM!

**Review by RiyaMehta2211 — COMMENTED** [review]
Like how you offered snippets of code in DG to offer more clarity on the JSON record object

**Pull request merged**